src/core/services/interimElement/interimElement.js

```diff
@@ -363,46 +380,76 @@ function InterimElementProvider() {
        *
        */
       function cancel(reason, options) {
```

Comment:
I would rather not make big changes in the PR, since it's already big enough. Could be an extra PR.